### DISCUSSION TITLE
training the modells

### DISCUSSION THREAD

**Discussion by MMSahin** [kyleskom/NBA-Machine-Learning-Sports-Betting#308] [General]
Hey,

I am currently working on a research project focusing on the application of machine learning in sports betting, and I came across this intriguing project.

I'm particularly interested in understanding the process of retraining the model and experimenting with different hyperparameters. Could someone please provide me with guidance on how to do this? I'm eager to explore the potential of this model and fine-tune it to better understand the objectives.

I appreciate any insights, recommendations, or resources you can share regarding retraining the modells. Your assistance will be invaluable for my research.

Thank you in advance for your support.

**Comment by nealmick** [comment]
My recommendation would be to a take look at the tensorflow docs, they have insanely cool projects.  Predicting the weather, survival rates on the titanic.  Just random stuff but it really helped me get a grasp of the underlying concepts. 
yess this one i remember:
https://www.tensorflow.org/tutorials/load_data/csv
but in general i would recommend playing with different datasets, and having fun with it!

lastly i will say in my experience, the data and how you arrange it is the most important thing by far.   Which data points and how representative they are of your target, and so on....

**Comment by kyleskom** [comment]
Hi @MMSahin you should be able to just run the python files from the Train-Models folder

**Reply by dexinthecity** [reply]
How often should we be running 

"# Create dataset with the latest data for 2022-23 season
cd src/Process-Data
python -m Get_Data
python -m Get_Odds_Data
python -m Create_Games

# Train models
cd ../Train-Models
python -m XGBoost_Model_ML
python -m XGBoost_Model_UO"

**Reply by kyleskom** [reply]
If you're just using the model daily, really never. These are there for more advanced work and creating / testing new models.

**Reply by y3z1n3** [reply]
does bot include data from games as they happen throughout the season, or do we have to input it manually?
thanks

**Reply by dexinthecity** [reply]
i just spent some time training new models, how do I run the new models to see how it is vs current model?

**Reply by eloco17** [reply]
Also wondering, does this automatically update database to include most recent games? Like will yesterdays be there? also does it retrain model?

**Reply by kyleskom** [reply]
That is not really how machine learning models work. It does not need to be retrained for every game.

**Reply by eloco17** [reply]
gotcha thanks. What do we need to run to update the data for recent games though? I can see the most recent game is from march 10th in the dataset_2012-23 db

**Reply by kyleskom** [reply]
What are you looking for with near real time updated data?

**Reply by eloco17** [reply]
nothing in particular. Teams change so I would want to know how to pull the data from this season once a bunch of games have been played. Or see how to automatically do this.

**Reply by kyleskom** [reply]
You can run the code in the `Process-Data` folder. Specifically in this order `get_data` , `get_odds_data` , `create_games`

Once you create the updated dataset with `create_games` you can use that to train models in the `Train-Models` folder.

**Reply by eloco17** [reply]
Thank you sir! Much appreciated for your promptness!!

**Reply by kyleskom** [reply]
You will need to edit the code in those files and change the parameters for the dates.

**Comment by dexinthecity** [comment]
> Hi @MMSahin you should be able to just run the python files from the Train-Models folder



> If you're just using the model daily, really never. These are there for more advanced work and creating / testing new models.

how would I go about seeing what the winning % after creating/training new models?

**Comment by STRATZ-Ken** [comment]
Imagine every time someone drove a tesla, we would retain a model. =D

**Comment by elfrost** [comment]
do we need to run the code in the Process-Data folder then train? or its good to go with this current 23-24 season?
in the csv nba-2023-UTC, do i need to put results in there?
